submit.py

```diff
@@ -1,15 +1,27 @@
 #!/usr/bin/env python
-import ConfigParser
+from __future__ import print_function
 import optparse
 import os
 import sys
 import itertools
-import mimetools
 import mimetypes
-import urllib
-import urllib2
 import traceback
-
+import random
+import string
+if sys.version_info[0] >= 3:
+    _PYTHON_VERSION = 3
+    import configparser
+    import urllib.parse
+    import urllib.request
+    import urllib.error
+else:
+    # Python 2, import modules with Python 3 names
+    _PYTHON_VERSION = 2
+    import ConfigParser as configparser
+    import urllib
+    import urllib2
+    urllib.request = urllib.error = urllib2
```

Comment:
On Friday, 15 January 2016, austrin <notifications@github.com> wrote:

> In submit.py
> <https://github.com/Kattis/kattis-cli/pull/3#discussion_r49827442>:
>
> > -
> > +import random
> > +import string
> > +if sys.version_info[0] >= 3:
> > +    _PYTHON_VERSION = 3
> > +    import configparser
> > +    import urllib.parse
> > +    import urllib.request
> > +    import urllib.error
> > +else:
> > +    # Python 2, import modules with Python 3 names
> > +    _PYTHON_VERSION = 2
> > +    import ConfigParser as configparser
> > +    import urllib
> > +    import urllib2
> > +    urllib.request = urllib.error = urllib2
>
> @mikaelgoldmann <https://github.com/mikaelgoldmann> does this rationale
> make sense to you (can we go ahead and merge), or do you want to change
> according to your suggestion?
>

Ignore my second suggestion.

It depends. If urllib2 provides a superset of both urllib.request and
urllib.error it's ok to merge. If not, I think this might be a misleading
approach.

>